### DISCUSSION TITLE
html attachment and send as pdf

### DISCUSSION THREAD

**Discussion by roxanne-tampus** [gophish/gophish#2695] [Q&A]
**Good day! Would like to know if possible to have a pdf attachment that was converted from html format when template variables is filled after starting a campaign?** 

html template
Hi {{.FirstName}}!

When campaign is started 

-- html file attachment > greetings_html.html
Hi John!

-- convert html to pdf and send to recipient as attachment> grettings_pdf.pdf
Hi John

**Comment by glennzw** [comment]
~Please open an issue, not a discussion.~

Actually this is more of a discussion point - we can't convert HTML to PDF, but it's an interesting idea. I'm not sure it's something we'd pursue though but you'd be welcome to create a fork and implement it yourself. I think you could modify the attachment.go switch statement to processes a custom extension (e.g .htpdf):

https://github.com/gophish/gophish/blob/a6627dfc6b1d5a27b7590ff4bf89950188dd39e2/models/attachment.go#L64 

After running the gophish template over the file you could use some third party library to convert the HTML to PDF.

Hope that helps!
